crates/utils/src/rate_limit/mod.rs

```diff
@@ -14,208 +14,142 @@ use std::{
+        state
+          .lock()
+          .expect("Failed to lock rate limit mutex for reading")
+          .remove_full_buckets(InstantSecs::now());
+      }
+    });
 
-    guard
-      .rate_limiter
-      .remove_older_than(duration, InstantSecs::now())
+    RateLimitCell { state }
   }
 
-  pub fn message(&self) -> RateLimitedGuard {
-    self.kind(RateLimitType::Message)
+  pub fn set_config(&self, config: EnumMap<ActionType, BucketConfig>) {
+    self
+      .state
+      .lock()
+      .expect("Failed to lock rate limit mutex for updating")
+      .set_config(config);
   }
 
-  pub fn post(&self) -> RateLimitedGuard {
-    self.kind(RateLimitType::Post)
+  pub fn message(&self) -> RateLimitChecker {
+    self.new_checker(ActionType::Message)
   }
 
-  pub fn register(&self) -> RateLimitedGuard {
-    self.kind(RateLimitType::Register)
+  pub fn post(&self) -> RateLimitChecker {
+    self.new_checker(ActionType::Post)
   }
 
-  pub fn image(&self) -> RateLimitedGuard {
-    self.kind(RateLimitType::Image)
+  pub fn register(&self) -> RateLimitChecker {
+    self.new_checker(ActionType::Register)
   }
 
-  pub fn comment(&self) -> RateLimitedGuard {
-    self.kind(RateLimitType::Comment)
+  pub fn image(&self) -> RateLimitChecker {
+    self.new_checker(ActionType::Image)
   }
 
-  pub fn search(&self) -> RateLimitedGuard {
-    self.kind(RateLimitType::Search)
+  pub fn comment(&self) -> RateLimitChecker {
+    self.new_checker(ActionType::Comment)
   }
 
-  pub fn import_user_settings(&self) -> RateLimitedGuard {
-    self.kind(RateLimitType::ImportUserSettings)
+  pub fn search(&self) -> RateLimitChecker {
+    self.new_checker(ActionType::Search)
   }
 
-  fn kind(&self, type_: RateLimitType) -> RateLimitedGuard {
-    RateLimitedGuard {
-      rate_limit: self.rate_limit.clone(),
-      type_,
+  pub fn import_user_settings(&self) -> RateLimitChecker {
+    self.new_checker(ActionType::ImportUserSettings)
+  }
+
+  fn new_checker(&self, action_type: ActionType) -> RateLimitChecker {
+    RateLimitChecker {
+      state: self.state.clone(),
+      action_type,
     }
   }
 }
 
+impl Default for RateLimitCell {
+  fn default() -> Self {
+    Self::new(enum_map! {
+      ActionType::Message => BucketConfig {
+        capacity: 180,
+        secs_to_refill: 60,
+      },
+      ActionType::Post => BucketConfig {
+        capacity: 6,
+        secs_to_refill: 300,
+      },
+      ActionType::Register => BucketConfig {
+        capacity: 3,
+        secs_to_refill: 3600,
+      },
+      ActionType::Image => BucketConfig {
+        capacity: 6,
+        secs_to_refill: 3600,
+      },
+      ActionType::Comment => BucketConfig {
+        capacity: 6,
+        secs_to_refill: 600,
+      },
+      ActionType::Search => BucketConfig {
+        capacity: 60,
+        secs_to_refill: 600,
+      },
+      ActionType::ImportUserSettings => BucketConfig {
+        capacity: 1,
+        secs_to_refill: 24 * 60 * 60,
+      },
+    })
+  }
+}
```

Comment:
This is only used in tests, so move it to a test module to prevent anyone from using it for prod code.

Reply:
I don't know how that would be possible, but I renamed `RateLimitCell::default` to `RateLimitCell::with_test_config` which might be good enough